### PR TITLE
rename doc to docs, increase version, add CI pipeline, rm old travis ci

### LINKED ISSUE #736: Update CI templates
Use just one CI pipeline with tests, coverage, pypi-releases and doc from the mirror in gitlab.

### PR DESCRIPTION
**Pull request by FWuellhorst**
This adds various CI features according to the latest templates.
Closes #736

### CODE REVIEW

**Comment by FWuellhorst** [comment]
The push mirror and the coverage still somehow fail due to an error in the coverage package, I will fix those later.

**Comment by FWuellhorst** [comment]
@DaJansenGit Coverage works now, the mirror should work as well. I guess the renamed old mirror was still updated, maybe a mixup in gitlab. I deleted the old mirror.

I added the changes from the master, should I merge this in the development? 

Other than that, feel free to review the changes :)

File: docs/source/conf.py [hunk 1]
```diff
@@ -30,7 +31,15 @@
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = ["sphinxcontrib.bibtex", "sphinx.ext.autodoc", "numpydoc"]
```
Comment (DaJansenGit) [inline docs/source/conf.py:33]: `sphinxcontrib.bibtex` not longer needed?

Comment (FWuellhorst) [reply docs/source/conf.py:33]: Some docs refer to some weird bib syntax, so I guess it is still needed.

**Comment by FWuellhorst** [comment]
I agree! @DaJansenGit: Should this go first in the dev? Or ->master and then master->dev?

**Comment by FWuellhorst** [comment]
@DaJansenGit can you approve?

**Review by DaJansenGit — APPROVED** [review]
There are still some changes and updates required for the documentation, but this will be documented and solved in #738

**Pull request merged**